public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/NewDimensionFields.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { SelectableValue, DataFrame, PanelData } from '@grafana/data';
+import { AccessoryButton, EditorList } from '@grafana/experimental';
+import { Select, HorizontalGroup, MultiSelect, Field } from '@grafana/ui';
+
+import { AzureMetricDimension, AzureMonitorOption, AzureMonitorQuery, AzureQueryEditorFieldProps } from '../../types';
+
+import { setDimensionFilters } from './setQueryValue';
+
+interface DimensionFieldsProps extends AzureQueryEditorFieldProps {
+  dimensionOptions: AzureMonitorOption[];
+}
+
+interface DimensionLabels {
+  [key: string]: Set<string>;
+}
+
+const useDimensionLabels = (data: PanelData | undefined, query: AzureMonitorQuery) => {
+  const [dimensionLabels, setDimensionLabels] = useState<DimensionLabels>({});
+  useEffect(() => {
+    let labelsObj: DimensionLabels = {};
+    if (data?.series?.length) {
+      // Identify which series' in the dataframe are relevant to the current query
+      const series: DataFrame[] = data.series.flat().filter((series) => series.refId === query.refId);
+      const fields = series.flatMap((series) => series.fields);
+      // Retrieve labels for series fields
+      const labels = fields
+        .map((fields) => fields.labels)
+        .flat()
+        .filter((item) => item!);
+      for (const label of labels) {
+        // Labels only exist for series that have a dimension selected
+        if (label) {
+          for (const [dimension, value] of Object.entries(label)) {
+            if (labelsObj[dimension]) {
+              labelsObj[dimension].add(value);
+            } else {
+              labelsObj[dimension] = new Set([value]);
+            }
+          }
+        }
+      }
+    }
+    setDimensionLabels((prevLabels) => {
+      const newLabels: DimensionLabels = {};
+      const currentLabels = Object.keys(labelsObj);
+      if (currentLabels.length === 0) {
+        return prevLabels;
+      }
+      for (const label of currentLabels) {
+        if (prevLabels[label] && labelsObj[label].size < prevLabels[label].size) {
```

Comment:
hey sorry do you mind explaining what is going on here? why are we only setting the bigger sized label

Reply:
I can explain this - we use the bigger sized label to ensure that we don't remove any of the potential dimension options. When the user chooses a dimension option the series will change and filter out the ones that don't match the condition. To ensure users can easily switch between dimensions we choose the larger labels object which will be unfiltered😊